creator-node/src/routes/files.js

```diff
@@ -246,6 +254,8 @@ const getDirCID = async (req, res) => {
       await findCIDInNetwork(queryResults.storagePath, CID, req.logger, libs)
       return await streamFromFileSystem(req, res, queryResults.storagePath)
     } catch (e) {
+      // Unset the cache-control header so that a bad response is not cached
+      res.removeHeader('cache-control')
```

Comment:
NICE

Reply:
not sure if i understand this correctly, but i dont think this is the right place for this? cause after it fails in this catch block, it goes to the ipfs fetch try/catch. only if it fails then we'd want to not cache right? cause this catch just logs, doesn't return anything to the client

Reply:
yeah i think dheeraj has a point, we should move this to the catch down below if we failed the IPFS cat

Reply:
a potential alternative would be to explicitly set when we know the write is successful. like in streamFromFileSystem or ipfsCat on the end event we can add the header like 
`.on('end', () => { res.setHeader('cache-control'...); res.end(); resolve() })`

and that would prevent unintentional caches

Reply:
^ i like that a lot

Reply:
@dmanjunath looks like your solution didn't work unf. for the stream case. Headers are already set on the first byte of the piping the stream to the res. It should still be ok here though because if there is an error, it should throw and be caught by the final catch (if it fails IPFS too)

Reply:
Want to move this to staging to test a bit more if ok w/ u

Reply:
Yeah go for it!